pkg/test/provisioner.go

```diff
@@ -77,6 +73,14 @@ func Provisioner(overrides ...ProvisionerOptions) *v1alpha5.Provisioner {
 		},
 		Status: options.Status,
 	}
+
+	if options.ProviderRef == nil {
+		if options.Provider == nil {
+			options.Provider = struct{}{}
+		}
+		provider, _ := json.Marshal(options.Provider)
```

Comment:
Shouldn't fail, but wouldn't hurt to `Expect(err).To(BeNil())`

Reply:
Trying to keep this decoupled from gomega. I'll panic instead.

Reply:
We're also totally ignoring the validation 😭 
`utilruntime.Must(provisioner.Validate(context.Background()))`